crates/mem-pool/src/pool.rs

```diff
@@ -939,10 +964,21 @@ impl MemPool {
         }
         // Handle state before txs
         // withdrawal
-        self.finalize_withdrawals(db, withdrawals.collect())?;
+        let withdrawals: Vec<WithdrawalRequest> = withdrawals.collect();
+        self.finalize_withdrawals(db, withdrawals.clone())?;
         // deposits
         let deposit_cells = self.pending_deposits.clone();
-        self.finalize_deposits(db, deposit_cells)?;
+        self.finalize_deposits(db, deposit_cells.clone())?;
+
+        // Fan-out next mem block to readonly node
+        if let Some(handler) = &self.fan_out_mem_block_handler {
+            handler.next_mem_block(
+                withdrawals,
+                deposit_cells,
+                txs.clone().collect(),
```

Comment:
Transactions here are not finalized in 'new' mem block, and you broadcast them twice.
They get broadcasted again in ```self.push_transaction_with_db()``` later.

Reply:
Nice catch! txs in next_mem_block should be removed.